src/main/java/application/FormController.java

```diff
@@ -117,20 +125,60 @@ public SubmitResponseBean submitForm(@RequestBody SubmitRequestBean submitReques
         FormSession formEntrySession = new FormSession(serializableFormSession);
         FormRecordProcessorHelper.processXML(formEntrySession.getSandbox(), formEntrySession.submitGetXml());
 
-        SubmitResponseBean submitResponseBean = new SubmitResponseBean(formEntrySession);
-
-        ResponseEntity<String> submitResponse =
-                submitService.submitForm(submitResponseBean.getOutput(),
-                        submitResponseBean.getPostUrl(),
-                        new DjangoAuth(authToken));
+        SubmitResponseBean submitResponseBean = validateSubmitAnswers(formEntrySession.getFormEntryController(),
+                formEntrySession.getFormEntryModel(),
+                submitRequestBean.getAnswers());
+
+        if(!submitResponseBean.getStatus().equals(Constants.SYNC_RESPONSE_STATUS_POSITIVE)
+                || !submitRequestBean.isPrevalidated()) {
+            submitResponseBean.setStatus(Constants.ANSWER_RESPONSE_STATUS_NEGATIVE);
+        } else {
+
+            ResponseEntity<String> submitResponse =
+                    submitService.submitForm(formEntrySession.getInstanceXml(),
+                            formEntrySession.getPostUrl(),
+                            new DjangoAuth(authToken));
+
+            if (submitResponse.getStatusCode().is2xxSuccessful()) {
+                sessionRepo.delete(submitRequestBean.getSessionId());
+            } else {
+                submitResponseBean.setStatus("error");
```

Comment:
i think if you don't send back a 200 response, the frontend will handle it as an error